runtime/src/bank/tests.rs

```diff
@@ -3413,7 +3413,7 @@ fn test_bank_pay_to_self() {
         .unwrap();
 }
 
-fn new_from_parent(parent: &Arc<Bank>) -> Bank {
```

Comment:
`Bank::new_from_parent()` takes an `Arc<Bank>`, not `&Bank`.
How can this compile? %)

Reply:
GitHub's reviews are confusing :(

When looking at this PR as a single change, your explanation makes sense.
And I am not sure if my question made any sense to you :)

But I was reviewing it commit by commit, you said you tried to break it into commits.
This is what I saw in the third commit, and this is where the question arose:

![image](https://github.com/solana-labs/solana/assets/694419/1852e2a8-47fb-4357-9204-0fb5fc8b25ec)

It seems impossible to construct a bank that takes `parent: &Bank`.
Maybe intermediate commits did not really compile.
Feel free to close this discussion, as when combined into a single diff, the PR does make sense.

Reply:
Ahh ok, yeah, thanks for explaining. I clearly did misunderstand your question as I was viewing the comments on PR's main page, which shows the collective diff.

> Maybe intermediate commits did not really compile.

Basically this. I believe I had the source code compiling in the first commit, and then I came back and got the test code compiling in two more commits. I probably should have provided more detailed commit messages and/or explicitly mentioned what was compiling after each intermediate commit.

Apologies for any wasted time and/or frustration this might have caused you